taipy/gui/utils/_evaluator.py

```diff
@@ -136,14 +140,15 @@ def __save_expression(
         expr_hash: t.Optional[str],
         expr_evaluated: t.Optional[t.Any],
         var_map: t.Dict[str, str],
+        lambda_expr: t.Optional[bool] = False,
     ):
         if expr in self.__expr_to_hash:
             expr_hash = self.__expr_to_hash[expr]
             gui._bind_var_val(expr_hash, expr_evaluated)
             return expr_hash
         if expr_hash is None:
             expr_hash = _get_expr_var_name(expr)
-        else:
+        elif not lambda_expr:
```

Comment:
Not too sure about how and what happens if `lambda_expr` indeed,,,

Reply:
if lambda expr, it has a hasname, we work with that

Reply:
Got it. May be a small comment stating so would help?